osu.Game/Online/Rooms/Room.cs

```diff
@@ -183,6 +194,8 @@ public void CopyFrom(Room other)
             EndDate.Value = other.EndDate.Value;
             UserScore.Value = other.UserScore.Value;
             QueueMode.Value = other.QueueMode.Value;
+            DifficultyRange.Value = other.DifficultyRange.Value;
+            PlaylistItemStats.Value = other.PlaylistItemStats.Value;
```

Comment:
`Room.DeepClone()` uses `CopyFrom()` to create a clone of the room, but then unsets some properties that shouldn't be copied across, and I'd argue these shouldn't get copied across in that case...? This in general is used in the "copy room" flow.

It's a bit of a weird one because that `DeepClone()` should really be named `CreateFreshCopy()` or something as it's ultra misleading right now...

Reply:
The only reason why `Room.DeepClone()` unsets `RoomID` is because that's used as an indicator for if the room has been created or not:
https://github.com/ppy/osu/blob/b0a74a68510b547a00e13233284e2de3b87a49a8/osu.Game/Screens/OnlinePlay/Match/RoomSubScreen.cs#L227-L245

As per https://github.com/ppy/osu/pull/16944#discussion_r812556260, I think this is fine as is unless they're causing a client-side issue.

Reply:
I've added an inline note about this one.